### PR TITLE
nixosTests.ldap: remove

### PR DESCRIPTION
**Pull request by flokli**
###### Motivation for this change
This tests seem to have worked in 15f105d41fb83bc93692702af96848e8887fc03c, but
broke somewhere in the meantime.

The current module doesn't seem to be generally usable and might need a
serious refactor. It requires quite some hacks to get it to work (see
https://github.com/NixOS/nixpkgs/issues/86305#issuecomment-621129942),
or how the test used systemd.services.openldap.preStart and made quite
some assumptions on internals.

On top of that, the test has been broken since 2020-01-12 (5 months), so
there's no guarantee this works at all.

Let's drop the test and throw a big deprecation warning to alert
everyone still using the module, and if nobody speaks up, remove it as well.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Ekleog** [comment]
As per https://github.com/NixOS/nixpkgs/issues/86305#issuecomment-624588814, the module works pretty well for basic usage, though maybe (I've never administered a complex LDAP server) it isn't adapted to complex use cases.

So IMO if we just deprecate/remove this without proposing anything better, we then lose the “simple” use case just working, and don't get the complex use cases working any better. I'm not sure what this would thus bring, though I totally understand that with the information available before my comment there it was the reasonable thing to do :)

**Comment by flokli** [comment]
As per https://github.com/NixOS/nixpkgs/issues/86305#issuecomment-624595960, if there's a more simpler way to test basic usage, such a VM test would help to verify (some) things are still working. 

Would you be willing to add a simple VM test on top?

**Comment by Mic92** [comment]
I can add a basic test based on my server: https://github.com/Mic92/dotfiles/blob/master/nixos/eve/modules/openldap.nix#L14

**Comment by flokli** [comment]
@Mic92 thanks! In that case, feel free to ping me on the PR :-)

File: nixos/modules/services/databases/openldap.nix [hunk 1]
```diff
@@ -242,6 +242,12 @@ in
       }
     ];
 
+    warnings = [ ''
+      The openldap module is unmaintained and staged for removal.
+      If you use it, please fix it and remove this message.
+      See https://github.com/NixOS/nixpkgs/issues/86305 for details.
```
Comment (Mic92) [inline nixos/modules/services/databases/openldap.nix:248]: This is not true. On my server I run openldap using nixpkgs-unstable and regulary update it.
If anything breaks I will fix it.

Comment (flokli) [reply nixos/modules/services/databases/openldap.nix:248]: Well, the module doesn't have a maintainers attribute currently, so I can't know ;-)

I'd personally think a simple VM test is more sustainable having you fixing breakages after-the fact.

For example, I planned refactoring some of the unit parameters, but couldn't verify things still work, as there was no passing test.

File: nixos/tests/all-tests.nix [hunk 2]
```diff
@@ -164,7 +164,6 @@ in
   kubernetes.rbac = handleTestOn ["x86_64-linux"] ./kubernetes/rbac.nix {};
   latestKernel.hardened = handleTest ./hardened.nix { latestKernel = true; };
   latestKernel.login = handleTest ./login.nix { latestKernel = true; };
-  ldap = handleTest ./ldap.nix {};
```
Comment (Mic92) [inline nixos/tests/all-tests.nix:167]: This is fine. We still have much simpler test called `openldap.nix`.

File: nixos/modules/services/databases/openldap.nix (hunk 1 above)
Comment (Mic92) [reply nixos/modules/services/databases/openldap.nix:248]: Add me as a maintainer than.

File: nixos/tests/all-tests.nix (hunk 2 above)
Comment (flokli) [reply nixos/tests/all-tests.nix:167]: OK

**Comment by flokli** [comment]
@Mic92 I pushed a new commit that only removes the broken VM tests, and adds you as a maintainer to the `openldap` module. The `openldap.nix` vm test is left as it is (and was all the time)

File: nixos/modules/services/databases/openldap.nix (hunk 1 above)
Comment (flokli) [reply nixos/modules/services/databases/openldap.nix]: Done, thanks!

**Review by Mic92 — APPROVED** [review]

**Pull request merged**